File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []
    size = len(progresses)
    
    for index in range(size):
        remain = 100 - progresses[index]
        days.append(remain//speeds[index])
        
        if remain % speeds[index] > 0:
            days[index] += 1
    
    index = 0
    
    while index < size:
        today = days[index]
        task = 0
        
        while index < size and days[index] <= today:
            task += 1
            index += 1
        
        answer.append(task)
    
    return answer

#[7, 3, 9] -> (7일)7, 3 / (9일)9
#[5, 10, 1, 1, 20, 1] -> (5일)5 / (10일)10, 1, 1 / (20일)20, 1

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n = progresses.length;
        Queue<Integer> result = new ArrayDeque<>();
        Queue<Integer> queue = new ArrayDeque<>();
        int count = 0;
        
        for (int i = 0; i < n; i++) {
            int period = (100 - progresses[i]) % speeds[i] == 0 ? (100 - progresses[i]) / speeds[i] : (100 - progresses[i]) / speeds[i] + 1;
            if (queue.isEmpty()) {
                queue.offer(period);
                count = 1;
                continue;
            }
            if (queue.peek() >= period) {
                count++;
                continue;
            }
            result.offer(count);
            count = 1;
            queue.poll(); 
            queue.offer(period);
        }
        result.offer(count);
        
        int[] answer = new int[result.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = result.poll();
        }
        return answer;
    }
}
/*
기능 -> 진도 100%일 때 반영
progresses = 배포순 작업진도
speeds = 각 작업 개발 속도

93% 작업 -> 7% / 1% = 7일 걸림
30% 작업 -> 70% / 30% = 3일 걸림
55% 작업 -> 45% / 5% = 9일 걸림

1번 + 2번 배포 (7일) - 3번 배포 (9일)
=> [2, 1]

[9]
cnt = 1
result = [2, 1]
*/